### DISCUSSION TITLE
Unable to crawl after error

### DISCUSSION THREAD

**Discussion by GeekyMushroom** [TumblThreeApp/TumblThree#345] [Q&A (Question & Answer)]
I was attempting to download my liked posts after a specific date, and forgot to stop the crawl before putting my computer in hibernate.
After turning my computer back on in the morning, I got an "unexpected error, application state may be corrupted" pop up mentioning a blog I have never tried crawling on this computer before, rather I had previously crawled it on my old computer back in 2019. I attempted closing and reopening the program (had to use task manager to kill it as hitting X would just make the error pop up again), installing two updates since (the first one I had to manually install because it was that committed to having this error pop up over and over), restarting my system, and it still keeps popping up.

It has made it so I am unable to crawl at all, when I hit download the button greys out but nothing happens aside from pause and stop becoming available, but if I hit either of those they grey out as well as leaving download greyed out.

I am not sure if this error is related but it is also giving me an error along the top saying "Error 1: Could not load file in library: Invalid byte encoding."

Is there anything I can do to fix this? I am also unsure of what information is needed from me so I will just paste the error message here

> Exception message
> 
> Could not find a part of the path 'F:\TumblThree\organized\WKM\Index\blueheartedmayor.tumblr'.
> 
> Exception type
> 
> System.IO.DirectoryNotFoundException
> 
> Callstack
> 
>    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
>    at TumblThree.Domain.Models.Blogs.Blog.SaveCore(String path) in C:\projects\Tumblthree\src\TumblThree\TumblThree.Domain\Models\Blogs\Blog.cs:line 1132
>    at TumblThree.Domain.Models.Blogs.Blog.SaveBlog() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Domain\Models\Blogs\Blog.cs:line 1128
>    at TumblThree.Domain.Models.Blogs.Blog.Save() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Domain\Models\Blogs\Blog.cs:line 1107
>    at TumblThree.Applications.Controllers.CrawlerController.Shutdown() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Controllers\CrawlerController.cs:line 88
>    at TumblThree.Applications.Controllers.ModuleController.Shutdown() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Controllers\ModuleController.cs:line 196
>    at TumblThree.Presentation.App.DoShutdown() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Presentation\App.xaml.cs:line 112
>    at TumblThree.Presentation.App.<RequestShutdown>b__7_1() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Presentation\App.xaml.cs:line 48
>    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
>    at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
> 
> Inner exception
> 
> -
> 
> TumblThree version
> 
> TumblThree 2.8.0.347 (64-bit)
> 
> Windows version
> 
> Windows 10 Home 64-bit, version 21H2 (10.0.19044.1889)
> 
> .Net Framework version
> 
> 4.0.30319.42000 64 Bit
> 
> .Net Version Support
> 
> 5 / 6
> 
> Default browser
> 
> Firefox (104.0.2)
> 
> Region settings
> 
> chr-Cher-US, US

**Comment by thomas694** [comment]
The `Exception type` in the popup tells that it's a `DirectoryNotFoundException` meaning that a part of the path `F:\TumblThree\organized\WKM\Index` cannot be found. Does this directory structure still exist, or is it e.g. an offline usb disk?

The second error "Invalid byte encoding" says that the encoding of one of the blog index files changed. 
> It could be that you opened one of the files in the Index folder in another program / text editor even if you did not (knowingly) save it there? At least one of the files has a UTF/Unicode format now which is not supported.
> 
> To find the file that makes the problems you can close TumblThree and move the files from the Index folder to a backup location. Then move each blog's files (e.g. abc.tumblr and abc_files.tumblr) back to the Index folder and start TumblThree afterwards. Close it again and repeat the step until TumblThree no longer loads the files, the last one added is the broken one. If you have many blogs, you can use small groups (e.g. 5 or 10 at once) to first narrow it down.
[#160]

Regarding the download problem, you can have a look in the log file (%LOCALAPPDATA%\TumblThree\TumblThree.log), whether you can find an error around the time you pressed the download button.

**Reply by GeekyMushroom** [reply]
regarding the directory it no longer is on the F: drive, it (along with all the other directories) is on my G: drive, and I am not sure where the index folder was.

I found an old version on a different drive and tried deleting all versions on my computer and reinstalling entirely (I realized I had way too many folders with different versions so this was the cleanest way to find the problem I think). This seems to have fixed some of the issues but I am still unable to crawl. Since the reinstall though it is actually giving me an error in the log now (it wasn't before), which is as follows

> 20220914 18:40:37.152	Err 	TumblrLikedByCrawler:CrawlPageAsync: Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: Cannot perform runtime binding on a null reference
>    at CallSite.Target(Closure , CallSite , Object )
>    at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
>    at TumblThree.Applications.Crawler.TumblrLikedByCrawler.ExtractPosts(String document) in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Crawler\TumblrLikedByCrawler.cs:line 366
>    at TumblThree.Applications.Crawler.TumblrLikedByCrawler.<CrawlPageAsync>d__16.MoveNext() in C:\projects\Tumblthree\src\TumblThree\TumblThree.Applications\Crawler\TumblrLikedByCrawler.cs:line 190

It repeated this error 4 times before stopping.
I am not sure what it means by C:\projects, that isn't a folder that exists on my machine. However, I am no longer getting the application state may be corrupted error which is good.

**Comment by thomas694** [comment]
The Index folder is a subfolder inside your download location that contains all the blog databases.

This error message says that the Liked-By crawler is currently broken.

These folders (C:\projects\...) shown in the error message are on the build machine where the app was compiled, not on your pc.
